pkg/spiffe/spiffe.go

```diff
@@ -76,17 +94,105 @@ func GenSpiffeURI(ns, serviceAccount string) (string, error) {
 func MustGenSpiffeURI(ns, serviceAccount string) string {
 	uri, err := GenSpiffeURI(ns, serviceAccount)
 	if err != nil {
-		log.Debug(err.Error())
+		spiffeLog.Debug(err.Error())
 	}
 	return uri
 }
 
 // GenCustomSpiffe returns the  spiffe string that can have a custom structure
 func GenCustomSpiffe(identity string) string {
 	if identity == "" {
-		log.Error("spiffe identity can't be empty")
+		spiffeLog.Error("spiffe identity can't be empty")
 		return ""
 	}
 
 	return URIPrefix + GetTrustDomain() + "/" + identity
 }
+
+// RetrieveSpiffeBundleRootCertsFromStringInput retrieves the trusted CA certificates from a list of SPIFFE bundle endpoints.
+// It can use the system cert pool and the supplied certificates to validate the endpoints.
+// The input endpointTuples should be in the json format of:
+//		{"map": {"foo": "URL1", "bar": "URL2"}}
+func RetrieveSpiffeBundleRootCertsFromStringInput(inputString string, extraTrustedCerts []*x509.Certificate) (
+	map[string][]*x509.Certificate, error) {
+	spiffeLog.Infof("Processing SPIFFE bundle configuration: %v", inputString)
+	var converted spiffeConfig
+	err := json.Unmarshal([]byte(inputString), &converted)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarchalling the input: %v", err)
+	}
+	return RetrieveSpiffeBundleRootCerts(converted, extraTrustedCerts)
+}
+
+// RetrieveSpiffeBundleRootCerts retrieves the trusted CA certificates from a list of SPIFFE bundle endpoints.
+// It can use the system cert pool and the supplied certificates to validate the endpoints.
+func RetrieveSpiffeBundleRootCerts(config spiffeConfig, extraTrustedCerts []*x509.Certificate) (
+	map[string][]*x509.Certificate, error) {
+	httpClient := &http.Client{}
+	caCertPool, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get SystemCertPool: %v", err)
+	}
+	for _, cert := range extraTrustedCerts {
+		caCertPool.AddCert(cert)
+	}
+
+	ret := map[string][]*x509.Certificate{}
+	for trustdomain, endpoint := range config.Map {
+		if !strings.HasPrefix(endpoint, "https://") {
+			endpoint = "https://" + endpoint
+		}
+		u, err := url.Parse(endpoint)
+		if err != nil {
+			return nil, fmt.Errorf("failed to split the SPIFFE bundle URL: %v", err)
+		}
+
+		config := &tls.Config{
+			ServerName: u.Hostname(),
+			RootCAs:    caCertPool,
+		}
+
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: config,
+		}
+
+		resp, err := httpClient.Get(endpoint)
```

Comment:
Do we need a retry here if there is a networking issue that causes the request fail?.